src/gui/rss/articlelistwidget.cpp

```diff
@@ -55,6 +55,7 @@ void ArticleListWidget::setRSSItem(RSS::Item *rssItem, bool unreadOnly)
 {
     // Clear the list first
     clear();
+    m_rssArticleToListItemMapping.clear();
```

Comment:
I usually use `void checkInvariant()` because when checking fail I want to know which condition is not satisfied. So, in this case `checkInvariant()` contains a sequence of assertions. In a harder case, I write in a way that it to reports all failed conditions to a log and then `throw` or `assert()`.